safe_transaction_service/utils/utils.py

```diff
@@ -8,6 +8,24 @@
 from gevent.monkey import saved
 
 
+class FixedSizeDict(dict):
+    """
+    Fixed size dictionary to be used as an LRU cache
+
+    Dictionaries are guaranteed to be insertion sorted from Python 3.7 onwards
+    """
+
+    def __init__(self, *args, maxlen=0, **kwargs):
+        self._maxlen = maxlen
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        dict.__setitem__(self, key, value)
```

Comment:
This could be after the maxlen condition because here we are adding one element more than defined size for a while.

Reply:
Good catch, but I don't think this is critical 😉